agent/rpc/peering/stream_test.go

```diff
@@ -147,10 +175,32 @@ func TestStreamResources_Server_LeaderBecomesFollower(t *testing.T) {
 	err2 := client.Send(input2)
 	require.NoError(t, err2)
 
+	// expect error
 	msg2, err2 := client.Recv()
 	require.Nil(t, msg2)
 	require.Error(t, err2)
 	require.EqualError(t, err2, "rpc error: code = FailedPrecondition desc = node is not a leader anymore; cannot continue streaming")
+
+	// expect a status error
+	st, ok := status.FromError(err2)
+	require.True(t, ok, "need to get back a grpc status error")
+	deets := st.Details()
+	// expect a LeaderAddress message
+	{
+		found := false
+		for _, deet := range deets {
+			la, ok := deet.(*pbpeering.LeaderAddress)
+
+			if !ok {
+				continue
+			} else {
+				found = true
+				require.Equal(t, "expected:address", la.Address)
+			}
+		}
+
+		require.True(t, found, "expected to find a LeaderAddress message in error details")
+	}
 }
```

Comment:
Same as above - would be nice to give an expected slice of elements and compare rather than this verbose loop